### PR TITLE
Adding initial base HTML partial, CSS

### PR DESCRIPTION
**Pull request by jonathanpli**
Added Pebble Extension to support global variables for templates
Added common.html partial for all pages
Added CSS temporary SCSS for consistency
Added CSS to .gitignore because SCSS handles generation

### CODE REVIEW

File: src/main/java/com/jonathanpli/ExceptionNullApplication.java [hunk 1]
```diff
@@ -42,13 +42,13 @@ public static void main(String[] args) {
 
 	@Bean
 	public static PebbleEngine pebbleEngine() {
-		PebbleEngine.Builder builder = new PebbleEngine.Builder()
+		PebbleEngine.Builder builder = new PebbleEngine.Builder().extension(new PebbleExtension())
```
Comment (fru1tstand) [inline src/main/java/com/jonathanpli/ExceptionNullApplication.java]: For all chained methods on a builder, please make newlines. This helps readability.
```
new PebbleEngine.Builder()
    .extension(...)
    .loader(...)
```

File: src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.jonathanpli.exceptionnull.controller.helpers;
+
+import com.mitchellbosecke.pebble.extension.AbstractExtension;
+import java.util.Map;
+import java.util.HashMap;
+
+/**
+ * Created by Jonathan on 4/9/17.
```
Comment (fru1tstand) [inline src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java]: Please javadoc and include what the extension does/provides. Up to you to `@author`, but I personally don't add the tag.

File: src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+public class PebbleExtension extends AbstractExtension {
+    private Map<String, Object> globals;
+
+    public PebbleExtension() {
+        globals = new HashMap<>();
+    }
+
+    @Override
+    public Map<String, Object> getGlobalVariables() {
+        globals.put("paths", getPaths());
```
Comment (fru1tstand) [inline src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java]: Is this intentional? Every call to #getGlobalVariables() will put getPaths() into the globals map.

File: src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        return globals;
+    }
+
+    private String[][] getPaths() {
+        String[][] paths = {
```
Comment (fru1tstand) [inline src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java]: Consider using [ImmutableList.of(T...)](https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/ImmutableList.html#of-E-), and making into a constant.

File: src/main/resources/templates/common.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            {% block content %}{% endblock %}
+        </main>
+
+        <footer>
+            {% block footer %}
+                <p>Copyright DumplingLTD 2017</p>
+            {% endblock %}
+        </footer>
+    </body>
+</html>
```
Comment (fru1tstand) [inline src/main/resources/templates/common.html]: nit: Please add a newline at the end of files

File: src/main/resources/templates/index.html [hunk 6]
```diff
@@ -1,11 +1,8 @@
...
-	</body>
-</html>
+{% extends "common" %}
+
+{% block title %}{{ title }}{% endblock %}
+
+{% block content %}
+	<h1>{{ title }}</h1>
+	<p id="description">{{ description }}</p>
+{% endblock %}
```
Comment (fru1tstand) [inline src/main/resources/templates/index.html]: nit: newline

File: src/main/scss/styles.scss [hunk 7]
```diff
@@ -1,9 +1,72 @@
+$primary-font: "Helvetica Neue", "Open Sans", sans-serif;
```
Comment (fru1tstand) [inline src/main/scss/styles.scss]: Please create a new scss partial file named "_variables.scss" and `@import 'variables'` at the top of this file. Move these variables into that file.

File: src/main/scss/styles.scss [hunk 8]
```diff
@@ -1,9 +1,72 @@
...
 	padding: 0;
 	margin: 0;
-	background-color: red;
+	background-color: $background-color;
+	color: $text-color;
+	font-family: $primary-font;
+}
+
+header {
+	border-bottom: 1px solid #C3C3C3;
```
Comment (fru1tstand) [inline src/main/scss/styles.scss]: Colors should never be hard coded. Use variables instead.

File: src/main/scss/styles.scss [hunk 9]
```diff
@@ -1,9 +1,72 @@
...
+			text-decoration: none;
+			padding: 10px 5px;
+			color: $text-color;
+		}
+	}
+}
+
+#description {
+	text-align: center;
+}
```
Comment (fru1tstand) [inline src/main/scss/styles.scss]: nit: newline

File: src/main/resources/templates/common.html [hunk 10]
```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
```
Comment (fru1tstand) [inline src/main/resources/templates/common.html:2]: We should comment template files with their function. For example "The main entrypoint used as a basis for all pages"

File: src/main/resources/templates/index.html [hunk 11]
```diff
@@ -1,11 +1,8 @@
...
-		<link rel="stylesheet" href="/css/styles.css" />
-	</head>
-	<body>
-		<div>Hiasdadsasssssssdsds</div>
-		<div>pre {{example}} post</div>
-		no
-		<pre>{{profiles}}</pre>
-	</body>
-</html>
+{% extends "common" %}
```
Comment (fru1tstand) [inline src/main/resources/templates/index.html]: Please comment and note if this is a directly accessible page, or another template file.

File: src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+
+import com.mitchellbosecke.pebble.extension.AbstractExtension;
+import java.util.Map;
+import java.util.HashMap;
+
+/**
+ * Created by Jonathan on 4/9/17.
+ */
+public class PebbleExtension extends AbstractExtension {
+    private Map<String, Object> globals;
```
Comment (fru1tstand) [inline src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java]: Consider using [ImmutableMap<K, V>](https://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/ImmutableMap.html).

**Review by fru1tstand — CHANGES_REQUESTED** [review]

File: src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java (hunk 4 above)
Comment (jonathanpli) [reply src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java]: Leaving as is for now, as Pebble's "for" loop isn't working as expected (as specified in documentation). Will figure out solution for future commit

File: src/main/resources/templates/common.html (hunk 10 above)
Comment (jonathanpli) [reply src/main/resources/templates/common.html:2]: Added

File: src/main/resources/templates/common.html (hunk 5 above)
Comment (jonathanpli) [reply src/main/resources/templates/common.html]: Added

File: src/main/resources/templates/index.html (hunk 6 above)
Comment (jonathanpli) [reply src/main/resources/templates/index.html]: Added

File: src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java (hunk 4 above)
Comment (fru1tstand) [reply src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java]: Please then add a `// TODO(jonathan): ...`

File: src/main/scss/styles.scss [hunk 13]
```diff
@@ -1,6 +1,4 @@
-$primary-font: "Helvetica Neue", "Open Sans", sans-serif;
-$background-color: #0E0B16;
-$text-color: #FFF;
+@import '_variables';
```
Comment (fru1tstand) [inline src/main/scss/styles.scss]: No need to include the underscore when [importing](http://sass-lang.com/guide#topic-5).

File: src/main/scss/styles.scss [hunk 14]
```diff
@@ -20,15 +18,15 @@ html, body {
 }
 
 header {
-	border-bottom: 1px solid #C3C3C3;
+	border-bottom: 1px solid $gray-border;
```
Comment (fru1tstand) [inline src/main/scss/styles.scss]: Up to you, feel free to do the following in _variables.scss:
```
$color-accent: <the grey color>;

$header-border-color: $color-accent;
$footer-border-color: $color-accent;
```
And modify the variable usage in this file accordingly.

**Review by fru1tstand — CHANGES_REQUESTED** [review]
Good. Please be sure to respond to all comments made. If it's a simple fix, feel free to just respond "done".

File: src/main/scss/styles.scss (hunk 14 above)
Comment (jonathanpli) [reply src/main/scss/styles.scss]: done

File: src/main/scss/styles.scss (hunk 13 above)
Comment (jonathanpli) [reply src/main/scss/styles.scss]: done

File: src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java (hunk 4 above)
Comment (jonathanpli) [reply src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java]: done

**Review by fru1tstand — COMMENTED** [review]
Please be sure to review and respond to all "outdated" comments. They're only visible on the "Conversation" tab for some reason.

File: src/main/java/com/jonathanpli/ExceptionNullApplication.java (hunk 1 above)
Comment (jonathanpli) [reply src/main/java/com/jonathanpli/ExceptionNullApplication.java]: done

File: src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java (hunk 2 above)
Comment (jonathanpli) [reply src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java]: done

File: src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java (hunk 12 above)
Comment (jonathanpli) [reply src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java]: done

File: src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java (hunk 3 above)
Comment (jonathanpli) [reply src/main/java/com/jonathanpli/exceptionnull/controller/helpers/PebbleExtension.java]: done

**Review by fru1tstand — APPROVED** [review]

**Pull request merged**